experimental/auth-logic/provenance_build_wrapper.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authlogic contains logic and tests for interfacing with the
+// authorization logic compiler
+package authlogic
+
+import (
+	"fmt"
+
+	"github.com/project-oak/transparent-release/common"
+	"github.com/project-oak/transparent-release/slsa"
+)
+
+type provenanceBuildWrapper struct{ provenanceFilePath string }
+
+func (pbw provenanceBuildWrapper) EmitStatement() UnattributedStatement {
+	handleErr := func(err error) {
```

Comment:
I think this is not very conventional. Also, I am not sure it is a good idea to panic. Is not better to return the error?

Reply:
+1 for canonical error handling and not panic

Reply:
https://github.com/project-oak/transparent-release/issues/40